### DISCUSSION TITLE
About "driverRotation"

### DISCUSSION THREAD

**Discussion by stc1988** [Moddable-OpenSource/moddable#702] [Q&A] [answered]
Regarding commits https://github.com/Moddable-OpenSource/moddable/commit/715475baedc5cb5568c62c650fe10fa3a003d490 and https://github.com/Moddable-OpenSource/moddable/commit/b9fd6bbf4ac341f6c1090b9c70dc0b453f410253, default rotation changed to `0` from `90` in M5stack.
So, the rotation of the screen has changed from before.

 How to use "driverRotation" and "rotation"?

**Comment by mkellner** [comment]
Hello. There is a blog post: [Run-time Display Rotation](https://blog.moddable.com/blog/rotate/).

**Reply by meganetaaan** [reply]
I think the value of `driverRotation` should be 90. This means that the normal orientation for the M5Stack is with the button at the bottom as many users recognize. And this (default 90 degrees rotation) is also done by [the official implementation of M5Stack driver](https://github.com/m5stack/M5Stack/blob/b4021226b8fd771fd43d5bad58595165da169e09/src/M5Display.cpp#L9). How do you think about it?

**Reply by meganetaaan** [reply]
- I had a mistake. The value is `driverRotation: 270` to make M5Stack correct rotation.
- And I realized that when both `driverRotation` and `rotation` are set to a non-zero value, `rotation` seems to be ignored. Is it expected?

**Comment by mkellner** [comment, marked as answer]
I will change to `driverRotation: 270` in the next drop.

And a note from the blog post above:

Warning: Don’t mix build-time and run-time rotation. Because not all display controllers support run-time rotation, the original Moddable SDK support for zero-overhead rotation performed at build-time remains available. These two forms of rotation are incompatible. Only use one of them in your project.
